src/docfx/build/xref/ExternalXRefSpec.cs

```diff
@@ -7,16 +7,18 @@
 
 namespace Microsoft.Docs.Build
 {
-    public class XrefSpec
+    internal class ExternalXrefSpec : IXrefSpec
     {
         public string Uid { get; set; }
 
+        public string Href { get; set; }
+
+        public Document ReferencedFile { get; set; }
```

Comment:
Since this class is still _THE_ data model, use `Document IXrefSpec.ReferencedFile => null` to hide this member both from serialization and from callers.

I would also like to prefix every data model with `Model` to clearly indicate that the class is a data contract and will be saved to disk, thus any changes to it must consider serialization and be backward compatible.

Reply:
Will hold the `Model` naming change here, we can update the data contracts once we agree on the naming

Reply:
yep